Taskfile.yml

```diff
@@ -324,42 +327,55 @@ tasks:
       # -race fails at the moment in gopter - possibly due to our shared generator variable?
       - go test -short -tags=noexit -timeout 10m -covermode atomic -coverprofile=controller-coverage.out -json -coverpkg="./..." -run '{{default ".*" .TEST_FILTER}}' ./... > '{{.TEST_OUT}}/controller-unit-tests.json'
 
-  controller:test-upgrade-pre:
-    desc: Test upgrading {{.CONTROLLER_APP}} and helm chart for new release
+  controller:test-upgrade-setup:
+    desc: Setup for upgrade test
     dir: "{{.CONTROLLER_ROOT}}"
     cmds:
-      - task: controller:kind-create
+      - task: controller:kind-create-wi
+      - task: controller:create-mi-for-workload-identity
       - task: controller:install-cert-manager
       - task: controller:docker-push-local
       - task: controller:gen-helm-manifest
       # We need the below to wait until cert-manager is up, otherwise the subsequent installation of webhooks fails. See https://cert-manager.io/next-docs/installation/verify/
       - "cmctl check api --wait=2m"
+
+  controller:test-upgrade-pre:
+    desc: Test upgrading {{.CONTROLLER_APP}} and helm chart for new release
+    dir: "{{.CONTROLLER_ROOT}}"
+    cmds:
       - "helm repo add asov2 https://raw.githubusercontent.com/Azure/azure-service-operator/main/v2/charts"
       - "helm repo update"
-      - "helm upgrade --install --devel aso2 asov2/azure-service-operator \
```

Comment:
Don't we need the `--devel` flag here? I have some recollection that we needed it for installs to actually work.

Reply:
https://helm.sh/docs/helm/helm_install/#options

It's for beta versions, since we're GA now we don't need it anymore.